LibCarla/source/carla/trafficmanager/LocalizationUtils.cpp

```diff
@@ -232,6 +233,43 @@ namespace LocalizationConstants {
     }
 
   }
+  SimpleWaypointPtr TrackTraffic::GetTargetWaypoint(Buffer& waypoint_buffer,const float& target_point_distance,uint64_t& index)
+  {
+    SimpleWaypointPtr target_waypoint = waypoint_buffer.front();
+    uint64_t  startPosn  = static_cast<uint64_t>(std::fabs(target_point_distance/SAMPLING_RESOLUTION));
+    bool mScanForward = false;
+    double target_point_dist_power = std::pow(target_point_distance, 2);
+
+    /// Condition to determine forward or backward scanning of  WayPoint Buffer.
+    if((startPosn < waypoint_buffer.size()) &&
```

Comment:
Separate out the condition (startPosn < waypoint_buffer.size()) as it as it's purpose is validation and not algorithmic.
If this condition fails, then our starting index is out of bounds of the container memory. If so, then just return the back most element of the waypoint_buffer as that would be the closest element to the caller's distance input.

Reply:
Implemented the comment

Reply:
Implemented